web/src/main/java/se/inera/intyg/webcert/web/service/utkast/UtkastServiceImpl.java

```diff
@@ -181,14 +181,20 @@ public int countFilterIntyg(UtkastFilter filter) {
     public Utkast createNewDraft(CreateNewDraftRequest request) {
 
         populateRequestWithIntygId(request);
-        request.setStatus(UtkastStatus.DRAFT_INCOMPLETE);
 
         String intygType = request.getIntygType();
 
         CreateNewDraftHolder draftRequest = createModuleRequest(request);
 
         String intygJsonModel = getPopulatedModelFromIntygModule(intygType, draftRequest);
 
+        if (intygJsonModel != null && request.getForifyllnad().isPresent()) {
```

Comment:
Föreslår att du gör en extract method refaktorisering och kapslar in logiken i privat metod med ett meningsfullt metodnamn. Det gör att createNewDraft-koden blir lite "renare" och mer lättläst.